modules/identity-provider-add.adoc

```diff
@@ -37,15 +48,50 @@ $ oc apply -f </path/to/CR>
 If a CR does not exist, `oc apply` creates a new CR and might trigger the following warning: `Warning: oc apply should be used on resources created by either oc create --save-config or oc apply`. In this case you can safely ignore this warning.
 ====
 
+ifndef::no-username-password-login[]
 . Log in to the cluster as a user from your identity provider, entering the
 password when prompted.
 +
 ----
 $ oc login -u <username>
 ----
+endif::no-username-password-login[]
+
+ifdef::no-username-password-login[]
+
+. Obtain a token from the OAuth server.
+
+.. You can request tokens at `<namespace_route>/oauth/token/request`. The value of `<namespace_route>` can be found using the following command:
++
+----
+$ oc get route oauth-openshift -n openshift-authentication -o json | jq .spec.host
+----
```

Comment:
You cannot do this if you haven't logged yourself in before :)

Instead, `oc login` should now prompt you with instructions how to log in, this is, however, conditional to admins doing the right thing and disabling the kubeadmin IdP (see https://bugzilla.redhat.com/show_bug.cgi?id=1671604).

Reply:
@stlaz So should I get rid of this step? And instead mention in the below step about trying `oc login` and having it provide instructions as long as the kubeadmin IDP is disabled?

However, the BZ shows this during verification:
```
# oc login
Login failed (401 Unauthorized)
Verify you have provided correct credentials.
You must obtain an API token by visiting https://<cluster-name>.devcluster.openshift.com/oauth/token/request
```
But wouldn't you need to be logged in already to access that page?